src/argilla/tasks/datasets/__main__.py

```diff
@@ -12,18 +12,59 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from typing import Optional
+
 import typer
 
 from argilla.tasks.callback import init_callback
+from argilla.tasks.datasets.delete import delete_dataset
 from argilla.tasks.datasets.list import list_datasets
 from argilla.tasks.datasets.push import push_to_huggingface
 
-app = typer.Typer(
-    help="Holds CLI commands for datasets management", invoke_without_command=True, callback=init_callback
-)
+_COMMANDS_REQUIRING_DATASET = ["delete", "push-to-huggingface"]
+
+
+def callback(
+    ctx: typer.Context,
+    name: Optional[str] = typer.Option(None, help="The name of the `FeedbackDataset` to which apply the command"),
+    workspace: Optional[str] = typer.Option(None, help="The name of the workspace where the `FeedbackDataset` belongs"),
+) -> None:
+    init_callback()
+
+    from argilla.client.feedback.dataset.local import FeedbackDataset
+
+    if ctx.invoked_subcommand not in _COMMANDS_REQUIRING_DATASET:
+        return
+
+    if name is None:
+        raise typer.BadParameter("The command requires a workspace name provided using '--name' option")
```

Comment:
Doesn't the plain `typer.Option` handle this for us if we set the default to `...` instead of `None`? Or is this because we want to show a custom exception to guide the user?

Reply:
in this case `name` is not a mandatory argument for all the commands like `datasets list`. In the first `if` we check if the executed command needs the `FeedbackDataset` (returns if don't). If this `if` condition executed gets evaluated, then the executed command needs the `FeedbackDataset` and the `name` option has not been provided.